mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -47,9 +47,13 @@ def __init__(
             )
         self._tdengine_connection_string = kwargs.get("connection_string")
         self.database = database
-        self._connection = self._create_connection()
+
+        self._connection = None
         self._init_super_tables()
 
+    def init(self):
+        self._connection = self._create_connection()
+
```

Comment:
I think that change is an obvious code smell, please mock what you need instead.

Reply:
I disagree. In my view, I/O doesn't belong in the constructor, and dependency injection is a common pattern to enable unit testing. The need for mocking is the smell.

Reply:
I could replace `init()` with lazy init (maybe in a property), though.

Reply:
> I/O doesn't belong in the constructor

Then we will probably not agree here :)
We both can find plenty of examples for it in mlrun and many other places.

This pattern, to be used in production according to this PR:
```py
a = A()
a.init()
```
Is:
1. Error-prone
2. Not natural in Python and ugly IMO

Note that you add all of this for a test, which is nice for plots, but doesn't test much TBH.

Reply:
Replaced with a property.

I would like the code to be testable for the purpose of adding more tests going forward.

Reply:
The issue with I/O in the constructor in this case (putting aside the general principle), is that it's not reasonable to have to mock out a bunch of I/O to be able to create the graph (a purely functional operation).